### DISCUSSION TITLE
How can I ensure my table sits on its own page?

### DISCUSSION THREAD

**Discussion by e05bf027** [quarto-dev/quarto-cli#7171] [Q&A]
### Description

I am using Quarto with R and RStudio to write a thesis document. I am adding a table that summarises the characteristics of patients I studied. I created this using `gtsummary`.

I have created the table and it renders correctly in my final pdf. However, I want to make it sit on it's own page rather than getting split across two pages. I don't know how to do this. 

Code to create table:

```
#| include: false
#| warning: false

demo_summary%>% 
  select(-c(patient_id,
            pathogenic_factor,
            ards_risk_factor,
            adm_date,
            # adm_location,
            dc_date_time,
            ventilator)) %>% 
  relocate(bmi, .after = weight_kg) %>% 
  tbl_summary(by = NULL,
              statistic = list(all_continuous() ~ "{mean} ({sd})",
                               all_categorical() ~ "{n} ({p}%)"),
              digits = list(all_continuous() ~ 1,
                            all_categorical() ~ 0),
              label = list(gender ~ 'Gender',
                           age_years ~ 'Age (years)',
                           height_cm ~ 'Height (cm)',
                           weight_kg ~ 'Weight (kg)',
                           bmi ~ 'BMI',
                           ards_type ~ 'ARDS classification',
                           adm_location ~ 'Admitting location',
                           los_days ~ 'LOS (days)',
                           apache_ii ~ 'Apache II',
                           total_proning_sessions ~ 'Proning sessions',
                           outcome ~ 'Outcome'),
              missing = 'ifany',
              missing_text = 'Not recorded') %>%
  bold_labels() %>% 
  # add_overall() %>% 
  as_gt() %>% 
  as_latex() -> demo_summary
```
I then use another chunk to print:

```
#| echo: false
#| tbl-cap-location: bottom
#| tbl-cap: Demographic characteristics of patients undergoing prone positioning

demo_summary
```

The caption renders at the top of the table, I am unsure why. Most frustratingly I want make sure the table isn't split across pages. I know how to achieve this with Latex, but am unsure how to achieve the same here.

Anyone any ideas?

**Comment by cscheid** [comment]
> The caption renders at the top of the table

This is a bug that we have fixed on our 1.4 pre-releases. What version are you using?

> I want make sure the table isn't split across pages.

Not splitting the table across pages will be a gt-specific setting, and I'm afraid I don't know the solution there.

You can try to figure out what's wrong by using `keep-tex: true`.

**Reply by e05bf027** [reply]
When I click on 'About RStudio' I get this:

> Version 2023.06.1+524 (2023.06.1+524)

**Reply by cscheid** [reply]
If you haven't installed a separate version of quarto, you probably are running 1.3. To fix the caption placement bug, you'd have to install a [pre-release version](https://quarto.org/docs/download/).

**Reply by e05bf027** [reply]
I installed the pre-release version. It seemed to proceed correctly, no warnings were thrown. However, i cannot access it. My OSX seems to simply not have it there. I have uninstalled and reinstalled. IU can't see any notifications in the security centre either.

I am running Max OSX 13.5 on an M1 Macbook Air. Is there any issue that often arises here?

**Reply by e05bf027** [reply]
Ignore the above, I got it installed. Unfortunately it throws errors elsewhere and won't render the document. I will wait for the full release.

**Reply by cderv** [reply]
> Unfortunately it throws errors elsewhere and won't render the document. 

Can you share more on this ? Maybe in a specific issue if you think this is something buggy in our pre-release. We value any feedback that makes our release better. We build pre-release almost daily so that users can install easily like you did, and we can find more problems before stable release. 

Just sharing this in case you have error that are not expected. Thank you

**Comment by andrewheiss** [comment]
I'm not sure there's a way (yet) to control table LaTeX positioning with {gt} (there's an open issue for it here https://github.com/rstudio/gt/issues/157).

For instance, this:

````qmd
```{r}
#| label: tbl-test
#| tbl-cap: "Testing"
data.frame(A = 1:2, B = 3:4) |>
  gt::gt()
```
````

emits this:

```latex
\begin{table}

\caption{\label{tbl-test}Testing}

\centering{

\begin{longtable*}{rr}
\toprule
A & B \\ 
\midrule\addlinespace[2.5pt]
1 & 3 \\ 
2 & 4 \\ 
\bottomrule
\end{longtable*}

}

\end{table}%
```

and there's not a way to inject options into the `\begin{table}` command like `\begin{table}[htbp]`
